tests/rptest/tests/partition_reassignments_test.py

```diff
@@ -547,7 +554,7 @@ def add_partition_during_inprogress_reassignment(
                 topic: str, count: int):
             assert try_add_partitions(
                 topic, count
-            ) is None, f'Expected failed add-partitions: topic {topic}, partition count {count}, output {out}'
+            ) is None or str, f'Expected reassignment in progress or success: topic {topic}, partition count {count}'
```

Comment:
This assert seems to always pass. It is parsed as `(try_add_partitions(topic, count) is None) or str`. `str` is a class, and a python class is an object of class `type`. The latter does not implement `__bool__(self)` or `__len__(self)` methods, so its instances are deemed true when used in a boolean context.

Perhaps you meant to check if the returned object is either None or an instance of `str`? `is` is not about checking the type, it's checking identity, not unlike `===` in other languages. So it'd be `res is None or isinstance(o, str)`.